src/rule-extractor/ObeysRuleExtractor.ts

```diff
@@ -0,0 +1,30 @@
+import { ProcessableElementDefinition } from '../processor';
+import { ExportableObeysRule } from '../exportable';
+import { getPath } from '../utils';
+
+export class ObeysRuleExtractor {
+  static process(input: ProcessableElementDefinition): ExportableObeysRule {
+    const invariantKeys: string[] = [];
+    if (input.constraint?.length > 0) {
+      input.constraint.forEach((constraint, i) => {
+        invariantKeys.push(constraint.key);
+        // The ObeysRule only contains the key, but these other attributes
+        // will be handled by the InvariantProcessor. So, they should not be
+        // used for CaretValueRules.
```

Comment:
Thinking about this again after leaving a long comment about where the invariant processing code should go (sorry) -- but if the invariant processing code were in the `StructureDefinition.extractInvariants` function, then as long as the `StructureDefinition.process` function called it _before_ the `StructureDefinition.extractRules` function, then `extractInvariants` could mark these paths as _processed_ instead of having to do it here.